scripts/_helpers.py

```diff
@@ -566,9 +570,19 @@ def mock_snakemake(rulename, root_dir=None, submodule_dir=None, **wildcards):
             if os.path.exists(p):
                 snakefile = p
                 break
+
+        if isinstance(configfile, str):
+            with open(configfile, "r") as file:
+                configfile = yaml.safe_load(file)
+        else:
```

Comment:
Rather than else, maybe "elif not isinstance(configfile, dict)" or drop the if case?
By default it is none, so the else case may not be necessary but a dictionary may be provided as well as input.
What do you think?

Almost ready to merge :D

Reply:
@davide-f, thanks for the feedback. I have initially meant to omit else case, because default if None. But then I thought what if the user gives something other than str, it can cause error. Therefore I have added additional else with None. Similar thing I have noticed in PyPSA-Eur. Maybe it is better then drop else case?

Reply:
> @davide-f, thanks for the feedback. I have initially meant to omit else case, because default if None. But then I thought what if the user gives something other than str, it can cause error. Therefore I have added additional else with None. Similar thing I have noticed in PyPSA-Eur. 

Let's drop it :) that function is not used by the normal user.
If provided a wrong input, probably it is safe for the code to through an error rather than silently fix it.
Great contribution!

Reply:
@davide-f, do you suggest dropping if condition fully and directly have:
```bash
            with open(configfile, "r") as file:
                configfile = yaml.safe_load(file)
```
If so, I will make those changes.

Reply:
I have dropped `else` condition, while keeping `if`, so to make sure, config is read only if the path is given as string.